### DISCUSSION TITLE
scimax, Zotero references + pdfs

### DISCUSSION THREAD

**Discussion by rmosvold** [jkitchin/scimax#446] [Q&A]
Hi! I have used Emacs for a long time, and am just starting to try out scimax, which I love so far! So thanks a lot for the wonderful work on this! There is just one thing I am not able to figure out, and I am not sure if it is a scimax thing or an org-ref thing — probably a little of both. 

After having learned from a blog post by Wouter Spekkink, I found a nice setup where a bibtex file of my Zotero references was automatically kept up-to-date (BetterBibtex), and org-ref was able to find my Zotero pdfs and open them automatically. After having installed scimax, the reference handling works much better, except that the pdfs are no longer found. I realize that scimax has a much more complex initiation file system than I have previously used, so I wonder where and how would be the best way to configure this. And if others have written about a good workflow for references + pdf handling with scimax, I would love to know about that too!

**Comment by jkitchin** [comment]
probably you are seeing a change from org-ref 2 to org-ref 3. maybe something changed on finding Zotero pdfs. Now pdfs are found via bibtex-completion functions. Maybe you set bibtex-completion-pdf-open-function to the function that worked before, and it will work.

**Reply by rmosvold** [reply]
You are probably right! Before, I don't think I had set a bibtex-completion-pdf-open-function, but I had set bibtex-completion-pdf-field to "file". Could that be it? If so, where in the scimax config-files is the right place to change this?

**Reply by rmosvold** [reply]
I had also copied the functions from Spekkink's blog post:

;; The function below allows me to consult the pdf of the citation I currently have my cursor on.
 (defun my/org-ref-open-pdf-at-point ()
  "Open the pdf for bibtex key under point if it exists."
  (interactive)
  (let* ((results (org-ref-get-bibtex-key-and-file))
         (key (car results))
         (pdf-file (funcall org-ref-get-pdf-filename-function key)))
    (if (file-exists-p pdf-file)
        (find-file pdf-file)
      (message "No PDF found for %s" key))))

  (setq org-ref-completion-library 'org-ref-ivy-cite
        org-export-latex-format-toc-function 'org-export-latex-no-toc
        org-ref-get-pdf-filename-function
        (lambda (key) (car (bibtex-completion-find-pdf key)))
        ;; See the function I defined above.
        org-ref-open-pdf-function 'my/org-ref-open-pdf-at-point
        ;; For pdf export engines.
        org-latex-pdf-process (list "latexmk -pdflatex='%latex -shell-escape -interaction nonstopmode' -pdf -bibtex -f -output-directory=%o %f")
        ;; I use orb to link org-ref, helm-bibtex and org-noter together (see below for more on org-noter and orb).
        org-ref-notes-function 'orb-edit-notes)

**Reply by rmosvold** [reply]
I have explored this a bit further, but still haven't figured out about the Zotero pdfs. Setting the bibtex-completion-library-path and the bibtex-completion-bibliography as suggested on the org-ref github page provides a setting that works and where pdfs are found, so perhaps I just have to use this as a nice opportunity to find a new (and possibly better) way of organizing my work :-)

Lots of things that I love about scimax, and I'll continue to explore it, so thanks again for all the wonderful work that is being done!!!!

**Comment by jkitchin** [comment]
If you have `bibtex-completion-pdf-field` set to "file" I think bibtex completion will look in the field for opening a pdf, and then in the bibtex-completion-library-path. This should be fine if Zotero puts an actual path in that field.

If it doesn't do that, you need a function that takes the path that is there and converts it to the zotero path (see `bibtex-completion-pdf-open-function`)

If that doesn't work, I guess you might have to write your own function that takes a key, and then computes the pdf path and opens it.
